rustls/src/crypto/mod.rs

```diff
@@ -588,6 +566,46 @@ pub fn default_fips_provider() -> CryptoProvider {
     aws_lc_rs::default_provider()
 }
 
+mod crypto_default_provider {
+    use alloc::sync::Arc;
+
+    #[cfg(not(feature = "std"))]
+    use alloc::boxed::Box;
+
+    #[cfg(not(feature = "std"))]
+    use once_cell::race::OnceBox;
+    #[cfg(feature = "std")]
+    use once_cell::sync::OnceCell;
+
+    use crate::crypto::CryptoProvider;
+
+    #[cfg(feature = "std")]
+    static PROCESS_DEFAULT_PROVIDER: OnceCell<Arc<CryptoProvider>> = OnceCell::new();
+    #[cfg(not(feature = "std"))]
+    static PROCESS_DEFAULT_PROVIDER: OnceBox<Arc<CryptoProvider>> = OnceBox::new();
```

Comment:
These `static`s should go below the usage site in the functions.

Reply:
Sounds like I should move these static declarations below all functions, to the very end of this new "default implementation" or "default impl" block. I would totally favor this. In case this is not what you are suggesting can you give me very explicit instructions wheee to and under what I should move these static declarations. Just hoping to avoid potential confusion on my part thanks!

Reply:
> Sounds like I should move these static declarations below all functions, to the very end of this new "default implementation" or "default impl" block.

Yup! I think that's exactly right.